src/content-script/in-page.ts

```diff
@@ -145,8 +145,12 @@ if (dappsWithoutCorrectEIP1193Support.has(window.location.origin)) {
 
 initializeEIP6963(provider, {
   onRequestProvider: () => {
-    // DApp supports EIP-6963
     pageObserver.stop();
+    dappDetection.registerEip6963SupportOnce(provider);
+  },
+  onAccessProvider: () => {
```

Comment:
Do we need to keep 2 different callbacks?

Reply:
They have different semantic, I think it's better to be truthful with intentions. Calling "requestProvider" one more time when it's not actually requested can be both unexpected and undesired